charts/turing/templates/turing-deployment.yaml

```diff
@@ -0,0 +1,130 @@
+      app: {{ template "turing.fullname" . }}
+      release: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+  template:
+    metadata:
+      labels:
+        app: {{ template "turing.fullname" . }}
+        release: {{ .Release.Name }}
+{{- if .Values.deployment.labels }}
+{{ toYaml .Values.deployment.labels | indent 8 -}}
+{{- end }}
+    spec:
+      {{- if .Values.clusterConfig.useInClusterConfig }}
+      serviceAccountName: {{ template "turing.serviceAccount.name" . }}
+      {{- end }}
+      {{- with (include "turing.initContainers" . | fromYaml) }}
+      {{- toYaml . | nindent 6 }}
+      {{- end }}
+      containers:
+      - name: api
+        image: {{ template "turing.image" . }}
+        imagePullPolicy: {{ .Values.deployment.image.pullPolicy }}
+        ports:
+        - containerPort: {{ .Values.service.internalPort }}
+        livenessProbe:
+          httpGet:
+            path: {{ .Values.deployment.livenessProbe.path }}
+            port: {{ .Values.service.internalPort }}
+            scheme: HTTP
+          initialDelaySeconds: {{ default "60" .Values.deployment.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ default "10" .Values.deployment.livenessProbe.periodSeconds }}
+          successThreshold: {{ default "1" .Values.deployment.livenessProbe.successThreshold }}
+          timeoutSeconds: {{ default "5" .Values.deployment.livenessProbe.timeoutSeconds }}
+        readinessProbe:
+          httpGet:
+            path: {{ .Values.deployment.readinessProbe.path }}
+            port: {{ .Values.service.internalPort }}
+            scheme: HTTP
+          initialDelaySeconds: {{ default "60" .Values.deployment.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ default "10" .Values.deployment.readinessProbe.periodSeconds }}
+          successThreshold: {{ default "1" .Values.deployment.readinessProbe.successThreshold }}
+          timeoutSeconds: {{ default "5" .Values.deployment.readinessProbe.timeoutSeconds }}
+        resources:
+          {{- toYaml .Values.deployment.resources | nindent 12 }}
+        args:
+        - -config
+        - /etc/turing/config.yaml
+        {{- if .Values.uiConfig }}
+        - -ui-config
+        - /etc/turing/ui.config.json
+        {{- end }}
+        {{- with .Values.deployment.extraArgs }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+        env:
+        {{- if index .Values "turing-postgresql" "enabled" }}
+        - name: DBCONFIG_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "common.postgres-password-secret-name" (list (index .Values "turing-postgresql") .Values.turingExternalPostgresql .Release .Chart ) }}
+              key: {{ include "common.postgres-password-secret-key" (list .Values.turingExternalPostgresql) }}
+        {{- else }}
+        - name: DBCONFIG_PASSWORD
+          value: {{ .Values.turingExternalPostgresql.password }}
```

Comment:
Creating a secret instead of setting a sensitive value in the env variable is better because everyone who can view the manifest will see the password. 

We have a few scenarios possible:
- use in-cluster postgres:
  1. create a secret with password
  2. source env variable from the new secret

- provide the password to external postgres:
  1. create a secret with password
  2. source env variable from the new secret

- provide secret with a password:
  1. source env variable from the secret provided

Currently, we use the third: create secrets outside of the chart and refer them in values

Reply:
Thanks for the detailed explanation, Maksym. I've updated this.